### PR TITLE
Absorption data

### PR DESCRIPTION
**Pull request by kashike**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/1476) | [**SpongeCommon**](https://github.com/SpongePowered/SpongeCommon/pull/1182)

### CODE REVIEW

**Review by gabizou — APPROVED** [review]
Good stuff.

File: src/main/java/org/spongepowered/common/data/manipulator/immutable/entity/ImmutableSpongeAbsorptionData.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Override
+    public ImmutableValue<Double> absorption() {
+        return this.getValueGetter();
+    }
+
+    @Override
+    protected ImmutableValue<Double> getValueGetter() {
+        return new ImmutableSpongeValue<>(this.usedKey, DataConstants.Entity.DEFAULT_ABSORPTION, this.getValue());
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/data/manipulator/immutable/entity/ImmutableSpongeAbsorptionData.java]: Should be cached.

**Review by gabizou — CHANGES_REQUESTED** [review]
Minor caching change. Need to have a test plugin to double check will work on players.

**Comment by stephan-gh** [comment]
@gabizou Test plugin is in the API PR.

**Pull request merged**